controller/tap/server.go

```diff
@@ -83,9 +84,11 @@ func (s *server) TapByResource(req *public.TapByResourceRequest, stream pb.Tap_T
 	log.Infof("Tapping %d pods for target: %+v", len(pods), *req.Target.Resource)
 
 	events := make(chan *public.TapEvent)
+	var wg sync.WaitGroup
 
 	go func() { // Stop sending back events if the request is cancelled
```

Comment:
I'm wondering if a better solution would be to get rid of this goroutine altogether and instead add a switch when reading from the events channel so that we read from both the events channel and the context's Done channel and return when Done becomes closed.  I think this has the intended behavior without needing to worry about closing the events channel.

Reply:
That is a good point. We would need to wrap the `select` in a infinite loop so that we don't exit the function in the event where we don't receive any events and the stream context isn't marked as `Done()`